### PR TITLE
postfixadmin: init at 3.3.9

### PR DESCRIPTION
**Pull request by globin**

### CODE REVIEW

File: nixos/modules/services/mail/postfixadmin.nix [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    hostName = mkOption {
+      type = types.str;
+      example = "postfixadmin.example.com";
+      description = "Hostname to use for the nginx vhost";
+    };
+
+    adminEmail = mkOption {
+      type = types.str;
+      example = "postfixadmin.example.com";
```
Comment (lheckemann) [inline nixos/modules/services/mail/postfixadmin.nix]:
```suggestion
      example = "postmaster@example.com";
```

File: nixos/modules/services/mail/postfixadmin.nix [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      description = ''
+        Define the Site Admin's email address below.
```
Comment (lheckemann) [inline nixos/modules/services/mail/postfixadmin.nix]:
```suggestion
        Defines the Site Admin's email address.
```

**Review by lheckemann — APPROVED** [review]
LGTM, some nits

**Comment by lheckemann** [comment]
>The database schema may change with every release, you should therefore run the 'setup.php' script when upgrading PostfixAdmin. setup.php which will take care of all database changes between revisions.

Should we do this in a oneshot service, or leave this up to people deploying postfixadmin using this module?

File: pkgs/servers/postfixadmin/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  installPhase = ''
+    mkdir $out
+    cp -r * $out/
+    ln -sf /etc/postfixadmin/config.local.php $out/
+    ln -sf /var/cache/postfixadmin/templates_c $out/
+  '';
+
+  meta = {
+    description = "Web based virtual user administration interface for Postfix mail servers";
```
Comment (lheckemann) [inline pkgs/servers/postfixadmin/default.nix:22]:
```suggestion
    description = "Web based virtual user administration interface for Postfix mail servers";
    homepage = "https://postfixadmin.sourceforge.io/";
```

**Comment by globin** [comment]
> > The database schema may change with every release, you should therefore run the 'setup.php' script when upgrading PostfixAdmin. setup.php which will take care of all database changes between revisions.
> 
> Should we do this in a oneshot service, or leave this up to people deploying postfixadmin using this module?

It needs the setup password which we don't have in plaintext, so this has to be done by the user.

**Pull request merged**